### ISSUE TITLE
execution of "squid -z" missing on web cache activation

### ISSUE THREAD

**Issue by renoow** [opnsense/core#2008]
Hello,
Maybe I'm wrong, but it looks like execution of squid -z on cache activation is missing.
Thanks for your work.
Regards,
Renoow.

**Comment by fichtner** [comment]
Hi Renoow,

It should be pulled in via setup.sh

https://github.com/opnsense/core/blob/master/src/opnsense/scripts/proxy/setup.sh

What issue are you seeing?

Cheers,
Franco

**Comment by fichtner** [comment]
Hmm, it’s not run on restart/reconfigure, maybe that is your issue. Initial setup ok, cache add and not working?

**Comment by renoow** [comment]
Hi Franco, 

When I check " Enable local cache (requires service restart) " squid restart hang. 

/var/log/squid.syslog.log 
OPNsense (squid-1): Failed to verify one of the swap directories, Check cache.log for details. Run 'squid -z' to create swap directories if needed, or if running Squid for the first time. 

/var/log/squid/cache.log 
2017/12/20 19:13:47 kid1| Squid Cache (Version 3.5.27): Exiting normally. 
2017/12/20 19:13:48 kid1| Set Current Directory to /var/squid/cache 
2017/12/20 19:13:48 kid1| Starting Squid Cache version 3.5.27 for amd64-portbld-freebsd11.0... 
2017/12/20 19:13:48 kid1| Service Name: squid 
2017/12/20 19:13:48 kid1| Process ID 46668 
2017/12/20 19:13:48 kid1| Process Roles: worker 
2017/12/20 19:13:48 kid1| With 57960 file descriptors available 
2017/12/20 19:13:48 kid1| Initializing IP Cache... 
2017/12/20 19:13:48 kid1| DNS Socket created at [::], FD 6 
2017/12/20 19:13:48 kid1| DNS Socket created at 0.0.0.0, FD 8 
2017/12/20 19:13:48 kid1| Adding domain localdomain from /etc/resolv.conf 
2017/12/20 19:13:48 kid1| Adding nameserver 127.0.0.1 from /etc/resolv.conf 
2017/12/20 19:13:48 kid1| Adding nameserver 192.168.122.1 from /etc/resolv.conf 
2017/12/20 19:13:48 kid1| Logfile: opening log stdio:/var/log/squid/access.log 
2017/12/20 19:13:48 kid1| Unlinkd pipe opened on FD 13 
2017/12/20 19:13:48 kid1| Local cache digest enabled; rebuild/rewrite every 3600/3600 sec 
2017/12/20 19:13:48 kid1| Logfile: opening log stdio:/var/log/squid/store.log 
2017/12/20 19:13:48 kid1| Swap maxSize 102400 + 262144 KB, estimated 28041 objects 
2017/12/20 19:13:48 kid1| Target number of buckets: 1402 
2017/12/20 19:13:48 kid1| Using 8192 Store buckets 
2017/12/20 19:13:48 kid1| Max Mem size: 262144 KB 
2017/12/20 19:13:48 kid1| Max Swap size: 102400 KB 
2017/12/20 19:13:48 kid1| ERROR: /var/squid/cache/00: (2) No such file or directory 
FATAL: Failed to verify one of the swap directories, Check cache.log 
for details. Run 'squid -z' to create swap directories 
if needed, or if running Squid for the first time. 
Squid Cache (Version 3.5.27): Terminated abnormally. 

Running "squid -z" manually and restarting squid solve the problem. 

I've tried on two fresh installs and obtain same result each time. 

Cordialement, 

Renaud MEDICI 
Inform'Ethic 
8, Route de Saint Julien - 07200 Ucel 
Tél. : +33 7 86 47 60 21 
Fax : +33 9 55 24 01 54 


De: "Franco Fichtner" <notifications@github.com> 
À: "opnsense/core" <core@noreply.github.com> 
Cc: "Renaud MEDICI" <rmedici@inform-ethic.fr>, "Author" <author@noreply.github.com> 
Envoyé: Mercredi 20 Décembre 2017 20:57:10 
Objet: Re: [opnsense/core] execution of "squid -z" missing on web cache activation (#2008) 



Hi Renoow, 

It should be pulled in via setup.sh 

[ https://github.com/opnsense/core/blob/master/src/opnsense/scripts/proxy/setup.sh | https://github.com/opnsense/core/blob/master/src/opnsense/scripts/proxy/setup.sh ] 


What issue are you seeing? 

Cheers, 
Franco

**Comment by renoow** [comment]
Yes, local cache activation activated when web cache already activated. 

Cordialement, 

Renaud MEDICI 
Inform'Ethic 
8, Route de Saint Julien - 07200 Ucel 
Tél. : +33 7 86 47 60 21 
Fax : +33 9 55 24 01 54 


De: "Franco Fichtner" <notifications@github.com> 
À: "opnsense/core" <core@noreply.github.com> 
Cc: "Renaud MEDICI" <rmedici@inform-ethic.fr>, "Author" <author@noreply.github.com> 
Envoyé: Mercredi 20 Décembre 2017 20:54:57 
Objet: Re: [opnsense/core] execution of "squid -z" missing on web cache activation (#2008) 



Hmm, it’s not run on restart/reconfigure, maybe that is your issue. Initial setup ok, cache add and not working?

**Comment by fichtner** [comment]
Aha, should be an easy fix. Will ping you tomorrow morning :)

**Comment by fichtner** [comment]
It looks like this is a problem in squid itself. If the service is running, squid -z -N says:

```
2017/12/21 06:49:36| Squid is already running!  Process ID 79343
```

and refuses to do the cache creation.

**Comment by fichtner** [comment]
@renoow you can try this prior to starting to configure squid:

    # opnsense-patch 25ca2ad 85c1e0a

However, the restart will always be necessary and I think manual stop + start already works.

**Comment by fichtner** [comment]
There is more issues here, I'm circling around to fix this for 18.1.

**Comment by fichtner** [comment]
This works now, but will not hit 17.7.x anymore.
